src/backend/distributed/relay/relay_event_utility.c

```diff
@@ -82,17 +82,50 @@ RelayEventExtendNames(Node *parseTree, char *schemaName, uint64 shardId)
 		{
 			AlterObjectSchemaStmt *alterObjectSchemaStmt =
 				(AlterObjectSchemaStmt *) parseTree;
-			char **relationName = &(alterObjectSchemaStmt->relation->relname);
-			char **relationSchemaName = &(alterObjectSchemaStmt->relation->schemaname);
+			ObjectType objectType = alterObjectSchemaStmt->objectType;
 
-			/* prefix with schema name if it is not added already */
-			SetSchemaNameIfNotExist(relationSchemaName, schemaName);
+			if (objectType == OBJECT_STATISTIC_EXT)
+			{
+				RangeVar *stat = makeRangeVarFromNameList(
+					(List *) alterObjectSchemaStmt->object);
+
+				/* set schema name and append shard id */
+				SetSchemaNameIfNotExist(&stat->schemaname, schemaName);
+				AppendShardIdToName(&stat->relname, shardId);
+
+				alterObjectSchemaStmt->object = (Node *) MakeNameListFromRangeVar(stat);
+			}
+			else
+			{
+				char **relationName = &(alterObjectSchemaStmt->relation->relname);
+				char **relationSchemaName =
+					&(alterObjectSchemaStmt->relation->schemaname);
+
+				/* prefix with schema name if it is not added already */
+				SetSchemaNameIfNotExist(relationSchemaName, schemaName);
+
+				/* append shardId to base relation name */
+				AppendShardIdToName(relationName, shardId);
+			}
 
-			/* append shardId to base relation name */
-			AppendShardIdToName(relationName, shardId);
 			break;
 		}
 
+#if PG_VERSION_NUM >= PG_VERSION_13
+		case T_AlterStatsStmt:
+		{
+			AlterStatsStmt *alterStatsStmt = (AlterStatsStmt *) parseTree;
+			RangeVar *stat = makeRangeVarFromNameList(alterStatsStmt->defnames);
+
+			AppendShardIdToName(&stat->relname, shardId);
+			SetSchemaNameIfNotExist(&stat->schemaname, schemaName);
```

Comment:
If I run:
``` 
ALTER STATISTICS foo.s1 SET STATISTICS 100;
``` 
We actually send:
``` 
SELECT worker_apply_shard_ddl_command (102039, 'public', 'ALTER STATISTICS foo.s1 SET STATISTICS 100')
``` 

The schema that is passed down via worker_apply_ddl_command always seems to refer to the relation's schema, not the statistics schema.

It probably does not matter much, since we actually always generate an explicit schema during deparsing, but maybe omit this call.